graph_snyk_cve_sync.py

```diff
@@ -267,131 +295,20 @@ def _parse_maven_data(self, maven_data):
-            for data in npm_data:
-                pkg = data['package']
-                logger.debug("Fetching details for package: {}".format(pkg))
-                try:
-                    versions = None
-                    if len(data['vulnerableVersions']) == 0:
-                        logger.info("False positive npm {} being ignored.".format(pkg))
-                        continue
-                    if pkg not in self.CVE_DATA['npm']:
-                        resp_obj = get_versions_and_latest_for_ep('npm', pkg)
-                        if resp_obj and (not isinstance(resp_obj, list)):
-                            versions = resp_obj['versions']
-                            self.CVE_DATA['npm'][pkg] = {
-                                "affected": [],
-                                "vulnerabilities": [],
-                                "all_ver": versions,
-                                "latest_version": resp_obj['latest_version'],
-                                "ecosystem": "npm",
-                                "package": pkg
-                            }
-                    else:
-                        versions = self.CVE_DATA['npm'][pkg]['all_ver']
-
-                    if versions:
-                        logger.info("Processing {}".format(data['id']))
-                        data['ecosystem'] = "npm"
-                        data['rules'] = self._get_version_rules(data['vulnerableVersions'])
-                        data['affected'] = self._get_affected_versions(data['rules'], versions)
-                        if len(data['affected']) == 0:
-                            logger.info("No active affected version found for {}. Ignored."
-                                        .format(pkg))
-                            continue
-                        total_vuln += 1
-                        del data['rules']
-                        self.CVE_DATA['npm'][pkg]['affected'].extend(data['affected'])
-                        self.CVE_DATA['npm'][pkg]['affected'] = list(
-                            set(self.CVE_DATA['npm'][pkg]['affected']))
-                        premium = str(data['premium']).lower() == 'true'
-                        data['pvtVuln'] = premium
-                        data['description'] = re.sub("[\'\"]", "", data['description'])
-                        self.CVE_DATA['npm'][pkg]['vulnerabilities'].append(data)
-                except ValueError:
-                    logger.error("Encountered an error while trying to fetch versions for "
-                                 "npm -> {}".format(pkg))
-        logger.info("NPM Data".center(50, '-'))
-        logger.info("Total affected packages: {}".format(len(self.CVE_DATA['npm'])))
-        logger.info("Total vulnerabilities: {}".format(total_vuln))
-        logger.debug(self.CVE_DATA['npm'])
+        logger.debug(self.CVE_DATA[eco])
 
     def _insert_cves(self):
         """Insert the cve data for each ecosystem."""
```

Comment:
Can these be generalised to avoid duplicates?
```
for ecosystem in ('maven', 'npm', 'pypi'):
   .. perform insertion for (ecosystem)
```

Reply:
Point noted.

Reply:
done